components/builder-api-client/src/lib.rs

```diff
@@ -517,6 +517,25 @@ impl Client {
         Ok(())
     }
 
+    /// Delete an origin
+    ///
+    ///  # Failures
+    ///
+    ///  * Remote builder is not available
+    ///  * Origin is populated with > 0 packages
+    ///  * Submitted Origin is not found
+    pub fn delete_origin(&self, origin: &str, token: &str) -> Result <()> {
+        let path = format!("depot/origins/{}", origin);
+
+        let res = self.add_authz(self.0.delete(&path), token).send()?;
+
+        if res.status != StatusCode::NoContent {
```

Comment:
If this is correct, it might warrant a comment. We *want* to see `StatusCode::NoContent` because we're trying to delete an empty origin, correct?

Alternatively, it might be better to structure the code like this:
```rust
        // We expect NoContent, because the origin must be empty to delete
        if res.status == StatusCode::NoContent {
            Ok(())
        } else {
            Err(err_from_response(res))
        }
```